### DISCUSSION TITLE
How to disable read for RTMP?

### DISCUSSION THREAD

**Discussion by abhinandan-chakraborty** [bluenviron/mediamtx#3983] [Questions]
### Question

Hi, I can disable RTMP completely by setting `rtmp: no` in the yaml file but I want to allow users to connect and publish through rtmp but not allow them to read it. Is it possible ?

**Comment by aler9** [comment]
Yes, it's possible if you enable HTTP-based authentication. Instantiate an HTTP server and create an endpoint that is able to receive HTTP POST requests and edit configuration accordingly:

```yml
authMethod: http
authHTTPAddress: http://localhost:8080/my-endpoint
```

The `/my-endpoint` will receive incoming requests in this format:

```
# {
#   "user": "user",
#   "password": "password",
#   "ip": "ip",
#   "action": "publish|read|playback|api|metrics|pprof",
#   "path": "path",
#   "protocol": "rtsp|rtmp|hls|webrtc|srt",
#   "id": "id",
#   "query": "query"
# }
```

You can then return status code `400` when `protocol == rtmp` and `action == read`, and status code `200` otherwise.
